### DISCUSSION TITLE
Frozen physics in Oculus browser VR

### DISCUSSION THREAD

**Discussion by IngoValente** [pmndrs/use-cannon#73] [General] [answered]
Somehow the physics stop working when in VR. I've never worked with service workers before, is there anything thats blocking them somehow? I have no idea how to debug this.
Example codesandbox here: https://mufpw.csb.app/

**Comment by stockhuman** [comment]
Could you try [this](https://codesandbox.io/s/use-cannon-vr-test-g7jq0?file=/src/index.js)? Unfortunately I don't have any VR hardware to test.

**Reply by IngoValente** [reply]
Not sure what changed (use-cannon version?), but unfortunately no luck. Still the same, as soon as I hit enter VR physics freeze.

**Reply by stockhuman** [reply]
Yes, I updated dependencies and changed a single line from useMemo to useEffect. Your console output could be helpful, if any.

**Comment by IngoValente** [comment]
For reference: I used an Oculus Quest with the oculus browser (based on chromium 81). Just tried it in Firefox Reality and there it works fine, so could be a bug in the browser.

**Reply by stockhuman** [reply]
Thanks for the context. :)

The fact that it works in another browser points more to this being an Oculus or Three issue. If it's not related to this recent change in the Oculus browser, this could be an implementation issue with react-three-fiber, although looking at the Canvas source, I don't immediately see how.

**Comment by IngoValente** [comment]
Ok some more testing. There's no console output unfortunately. Strange thing is that I just tried using cannon-es without use-cannon, and everything works fine... This is the working sandbox: https://m45m7.csb.app/
So looks like the oculus browser has some problems with a service worker.

**Comment by codynova** [comment, marked as answer]
Seems that this was actually caused by #99 and shouled be fixed by #100, which was released in `use-cannon@0.5.1`
